lib/get-commits.js

```diff
@@ -44,22 +44,28 @@ const getVersionHead = require('./get-version-head');
  * @return {Promise<Result>} The list of commits on the branch `branch` since the last release and the updated lastRelease with the gitHead used to retrieve the commits.
  *
  * @throws {SemanticReleaseError} with code `ENOTINHISTORY` if `lastRelease.gitHead` or the commit sha derived from `config.lastRelease.version` is not in the direct history of `branch`.
- * @throws {SemanticReleaseError} with code `ENOGITHEAD` if `lastRelease.gitHead` is undefined and no commit sha can be found for the `config.lastRelease.version`.
  */
 module.exports = async ({version, gitHead} = {}, branch, logger) => {
-  let gitTag;
-  if (gitHead || version) {
-    try {
-      ({gitHead, gitTag} = await getVersionHead(gitHead, version, branch));
-    } catch (err) {
-      if (err.code === 'ENOTINHISTORY') {
-        logger.error(notInHistoryMessage(err.gitHead, branch, version));
-      } else {
-        logger.error(noGitHeadMessage(branch, version));
+  if (gitHead) {
```

Comment:
Maybe add a comment referencing https://github.com/npm/read-package-json/pull/77 for explanation?

Reply:
It's not related to npm/read-package-json#77. The workaround for npm/read-package-json#77  has been moved to the npm plugin.

This is related to the `ENOTINHISTORY` error. That allow users to create a tag for a given commit in case they have rewritten their master history and the last release commit is missing.